pyomo/contrib/mindtpy/util.py

```diff
@@ -2,22 +2,27 @@
 #
 #  Pyomo: Python Optimization Modeling Objects
 #  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
-#  Under the terms of Contract DE-NA0003525 with National Technology and 
-#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
 #  rights in this software.
 #  This software is distributed under the 3-clause BSD License.
 #  ___________________________________________________________________________
 
 """Utility functions and classes for the MindtPy solver."""
 from __future__ import division
-
 import logging
 from pyomo.common.collections import ComponentMap
-from pyomo.core import Objective, Suffix
+from pyomo.core import (Block, Constraint,
+                        Objective, Reals, Suffix, Var, minimize, maximize, RangeSet, ConstraintList, TransformationFactory)
 from pyomo.core.expr import differentiate
 from pyomo.core.expr import current as EXPR
 from pyomo.opt import SolverFactory
 from pyomo.solvers.plugins.solvers.persistent_solver import PersistentSolver
+from pyomo.contrib.pynumero.interfaces.pyomo_nlp import PyomoNLP
```

Comment:
This unconditionally imports pynumero, which requires (optional) Pyomo dependencies (numpy and scipy).  We should leverage the deferred import functionality here:
```python
from pyomo.common.dependencies import attempt_import
# ...
pyomo_nlp = attempt_import('pyomo.contrib.pynumero.interfaces.pyomo_nlp')[0]
```
...then use `pyono_nlp.PyomoNLP` below.

Reply:
Thanks. Fixed.